python_ta/typecheck/base.py

```diff
@@ -129,42 +133,83 @@ def lookup_method(name, caller_type, *args):
     return TYPE_SIGNATURES[caller_origin][name]
 
 
+def make_set(node):
+    node.parent = node
```

Comment:
New nodes should have their `parent` set to `None`. This should also be done in the constructor of `TNode`. Never add new attributes outside of a constructor.